tests/related-files-test.el

```diff
@@ -38,15 +38,41 @@
   (should (safe-local-variable-p 'related-files-jumpers nil))
   (should-not (safe-local-variable-p 'related-files-jumpers (list (lambda (place) place)))))
 
+
+;;; Define test jumpers
+
+(defalias 'related-files-test-jumper-identity #'identity)
+
+(defun related-files-test-jumper-const (_place)
+  "Ignore arguments, return 'place."
+  'place)
+
+(defun related-files-test-jumper-atom (_place)
+  "Return \"/foo\"."
+  "/foo")
+
+(defun related-files-test-jumper-list (_place)
+  "Return (\"/bar1\" \"/bar2\")."
+  '("/bar1" "/bar2"))
+
+(defun related-files-test-jumper-singleton (_place)
+  "Return (\"/baz\")."
+  '("/baz"))
+
+(defalias 'related-files-test-jumper-nil #'ignore)
+
+(defvar related-files-test-jumper-args nil)
+
+(defun related-files-test-push-place-jumper (place)
+  (push place related-files-test-jumper-args)
```

Comment:
I would prefer avoiding tests with side effects. We might quickly get into trouble with tests only passing when executed in a certain order. What about defining this jumper with `cl-labels` or similar?

Reply:
> I would prefer avoiding tests with side effects.

I understand, and would usually agree. Strangely though, I've tried  `cl-labels` and `cl-flet`, each nested in a variety of ways in the test, and I just can't make it work. I'd welcome if you could, but at the moment I can't see another way of achieving this without side effects.

Reply:
Ok. Thanks for trying.